chore: add more documentation for configuring source maps properly (#6119)
cypress-io/cypress-documentation: docs/app/guides/debugging.mdx

By default, Cypress will include an inline source map in your spec file, so you
will get the most out of the error experience. If you
[modify the preprocessor](/api/node-events/preprocessors-api), ensure that inline
source maps are enabled to get the same experience. Some examples of this include:

- With webpack and the
  [webpack preprocessor](https://github.com/cypress-io/cypress/tree/develop/npm/webpack-preprocessor), set
  [the `devtool` option](https://webpack.js.org/configuration/devtool/) to
  `inline-source-map`.
- With esbuild and the 
  [esbuild preprocessor](https://github.com/bahmutov/cypress-esbuild-preprocessor/tree/main), set
  [the `sourcemap` option](https://esbuild.github.io/api/#sourcemap) to `inline` when 
  [creating the bundler](https://github.com/bahmutov/cypress-esbuild-preprocessor#esbuild-options).

> - Cucumber and the
>   [cucumber preprocessor](https://github.com/badeball/cypress-cucumber-preprocessor/tree/master)
>   rely on third party bundlers to bundle the code.

Author: ```suggestion
- With cucumber and the
  [cucumber preprocessor](https://github.com/badeball/cypress-cucumber-preprocessor/tree/master)
  third party bundlers are used to bundle the code.
```
